DEVELOPMENT.md

```diff
@@ -78,6 +78,18 @@ When enabled by default, this also works with editor plugins like
 
 [vim-fugitive]: https://github.com/tpope/vim-fugitive
 
+### Incremental Build
+
+Bazel is capable of doing incremental builds where it builds only subset of
+files that are impacted by file changes. To use that, just start TensorBoard
+with `ibazel` instead of `bazel`. If you do not have `ibazel` binary on your
+system, you can do use command below.
+
+```sh
+# After running `yarn` in the repo.
+yarn run ibazel run tensorboard -- --logdir [LOG_DIR]
```

Comment:
As the L89 indicates, you need to run `yarn` on the repo first.

Reply:
Ah yes, I had to run `yarn` again after checking out your changes.  I feel like this could be more explicit but perhaps it will not be a big deal.

So just to make sure I understand:
`yarn run ibazel run tensorboard -- --logdir [LOG_DIR]`
... would only need to be run once and afterward you would alway run:
`ibazel run tensorboard -- --logdir [LOG_DIR]?`

Or is it valuable to always include the `yarn run` part?

Reply:
Yes, you need to use `yarn run`. Just so we are on the same page, let me be explicit.

`yarn` is synonymous as `npm install` which installs all dependencies according to yarn-lock.json (npm uses some other name for shrinkwrap file). After I updated package.json and yarn-lock.json, you _must use_  `yarn` to update your local dependencies. Do note that this operation is only useful for VS Code and other tools and _is not_ used by Bazel. Bazel manages its own `node_modules` folder inside bazel-* directory.

You may have read somewhere that installing node packages globally is warn against. I personally do not make use of it because I do not make sure to keep them up-to-date and they can pose security threats. 

When installing packages, if a package.json contains `bin` property, it will symlink the entry point as specified in the value of `bin` to `./node_modules/.bin/[package_foo_bin]`.  `yarn run ibazel`, `npm run ibazel`, or `npx ibazel`, modulo some gotcha, is synonymous as `./node_modules/.bin/ibazel run tensorboard ...`. Like you indicated elsewhere, if you really really want `ibazel` binary, you may use `npm i -g @bazel/ibazel` but I do not want to recommend that to everyone for the reasons I mentioned above.

Reply:
(Drive-by +1 to the clear explanation—thank you!—and you can always
create a shell function or alias for `ibazel` to point to something like
`~/git/tensorboard/node_modules/...` rather than a global install.)

Reply:
Thanks for the explanation. Now I realize I was indeed relying on a global ibazel. I've uninstalled it.

I still have a problem with the command, though.  Relevant parts:
```shell
$ yarn run ibazel run tensorboard -- --logdir /google/data/ro/teams/tensorboard/logfiles/
yarn run v1.22.0
warning From Yarn 1.0 onwards, scripts don't require "--" for options to be forwarded. In a future version, any explicit "--" will be forwarded as-is to the scripts.
$ /usr/local/google/home/bdubois/git/tensorboard/node_modules/.bin/ibazel run tensorboard --logdir /google/data/ro/teams/tensorboard/logfiles/

<snip>

Error: A logdir or db must be specified. For example `tensorboard --logdir mylogdir` or `tensorboard --db sqlite:~/.tensorboard.db`. Run `tensorboard --helpfull` for details and examples.
```

Reply:
Argh. Fixed it.